lib/chef/knife/core/bootstrap_context.rb

```diff
@@ -40,7 +40,16 @@ def initialize(config, run_list, chef_config, secret = nil)
         end
 
         def bootstrap_environment
-          @chef_config[:environment] || '_default'
+          # Check for environment in CLI, first_boot_attributes, then in the
+          # config file. Use '_default' if no environment specified.
+
+          # @chef_config picks up @config's environment setting, even when
+          # environment is set in the config file being used. However,
+          # @config[:environment] is not polluted by @chef_config[:environment].
+          @config[:environment] ||
+            (@config[:first_boot_attributes][:environment] if @config[:first_boot_attributes]) ||
```

Comment:
That's an old alias: https://github.com/chef/chef/blob/master/lib/chef/node.rb#L133